### PR TITLE
Documentation & Samples: Add IDisposable to all examples & other improvements

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description

1. Add using statement to all FeedIterators in examples part of the comments
1. Add using statement to all FeedIterators in the samples
1. Add exception url to all async methods
1. Add additional examples to the database and container query stream iterator comments, and test to validate the new examples
1. Add more info the response.StatusCode to clarify it's usage.
 
## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/User/User.cs [hunk 1]
```diff
@@ -213,12 +159,13 @@ public abstract class User
...
         /// <param name="requestOptions">(Optional) The options for the user query request <see cref="QueryRequestOptions"/></param>
         /// <returns>An iterator to go through the permission</returns>
+        /// <exception>https://aka.ms/cosmosdb-dot-net-exceptions</exception>
         /// <example>
         /// 1. This create the type feed iterator for permission with queryText as input,
         /// <code language="c#">
         /// <![CDATA[
         /// string queryText = "SELECT * FROM c where c.status like 'start%'";
-        /// FeedIterator<PermissionProperties> feedIterator = this.users.GetPermissionQueryIterator<PermissionProperties>(queryText);
+        /// using (FeedIterator<PermissionProperties> feedIterator = this.users.GetPermissionQueryIterator<PermissionProperties>(queryText);
```
[the remark this answers is not in the document]
Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/Resource/User/User.cs:168]: Ditto on the other places where we only changed the first line

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosBasicQueryTests.cs [hunk 2]
```diff
@@ -304,7 +304,7 @@ public async Task ItemTest(bool directMode)
             List<dynamic> queryResults = await this.ToListAsync(
                   container.GetItemQueryStreamIterator,
                  container.GetItemQueryIterator<dynamic>,
-                 "select * from T where STARTSWITH(T.id, \"BasicQueryItem\")",
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosBasicQueryTests.cs:307]: Why is this required?

Comment (j82w) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosBasicQueryTests.cs:307]: It's just testing the new query feature where it ignores the case.

File: Microsoft.Azure.Cosmos/src/Resource/Scripts/Scripts.cs [hunk 3]
```diff
@@ -736,7 +706,7 @@ public abstract class Scripts
         /// <returns>
         /// A <see cref="Task"/> containing a <see cref="TriggerResponse"/> which wraps a <see cref="TriggerProperties"/> containing the updated resource record.
         /// </returns>
-        /// <exception cref="ArgumentNullException">If <paramref name="triggerProperties"/> is not set.</exception>
```
[the remark this answers is not in the document]
Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Resource/Scripts/Scripts.cs:739]: `ArgumentNullException` extends `ArgumentException` which is covered in the aka.ms. I doubt this is the only scenario that can cause an ArguementException which makes the current text misleading.

File: Microsoft.Azure.Cosmos/src/Handler/TransportHandler.cs [hunk 4]
```diff
@@ -31,14 +31,6 @@ public TransportHandler(CosmosClient client)
             RequestMessage request,
             CancellationToken cancellationToken)
         {
-            string stackTrace = new StackTrace().ToString();
```
[the remark this answers is not in the document]
Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Handler/TransportHandler.cs]: Yes, should be gone now with syncing to master.

File: Microsoft.Azure.Cosmos/src/Resource/Scripts/Scripts.cs [hunk 5]
```diff
@@ -795,27 +766,7 @@ public abstract class Scripts
         /// <param name="requestOptions">(Optional) The options for the user defined function request.</param>
         /// <param name="cancellationToken">(Optional) <see cref="CancellationToken"/> representing request cancellation.</param>
         /// <returns>A task object representing the service response for the asynchronous operation.</returns>
-        /// <exception cref="ArgumentNullException">If <paramref name="userDefinedFunctionProperties"/> is not set.</exception>
```
[the remark this answers is not in the document]
Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Resource/Scripts/Scripts.cs:798]: Please see the other comment. No reason to duplicate response.
